src/Models/HydrostaticFreeSurfaceModels/SplitExplicitFreeSurfaces/barotropic_split_explicit_corrector.jl

```diff
@@ -1,39 +1,51 @@
 # Kernels to compute the vertical integral of the velocities
-@kernel function _barotropic_mode_kernel!(U, V, grid, ::Nothing, u, v)
+@kernel function _barotropic_mode_kernel!(U̅, V̅, grid, ::Nothing, u, v, η)
     i, j  = @index(Global, NTuple)
-    barotropic_mode_kernel!(U, V, i, j, grid, u, v)
+    barotropic_mode_kernel!(U̅, V̅, i, j, grid, u, v, η)
 end
 
-@kernel function _barotropic_mode_kernel!(U, V, grid, active_cells_map, u, v)
+@kernel function _barotropic_mode_kernel!(U̅, V̅, grid, active_cells_map, u, v, η)
     idx = @index(Global, Linear)
     i, j = active_linear_index_to_tuple(idx, active_cells_map)
-    barotropic_mode_kernel!(U, V, i, j, grid, u, v)
+    barotropic_mode_kernel!(U̅, V̅, i, j, grid, u, v, η)
 end
 
-@inline function barotropic_mode_kernel!(U, V, i, j, grid, u, v)
-    @inbounds U[i, j, 1] = Δzᶠᶜᶜ(i, j, 1, grid) * u[i, j, 1]
-    @inbounds V[i, j, 1] = Δzᶜᶠᶜ(i, j, 1, grid) * v[i, j, 1]
+@inline function barotropic_mode_kernel!(U̅, V̅, i, j, grid, u, v, η)
+    k_top  = size(grid, 3) + 1
+
+    hᶠᶜ = static_column_depthᶠᶜᵃ(i, j, grid)
+    hᶜᶠ = static_column_depthᶜᶠᵃ(i, j, grid)
+
+    Hᶠᶜ = dynamic_column_depthᶠᶜᵃ(i, j, k_top, grid, η)
+    Hᶜᶠ = dynamic_column_depthᶜᶠᵃ(i, j, k_top, grid, η)
+
+    σᶠᶜ = ifelse(hᶠᶜ == 0, one(grid), Hᶠᶜ / hᶠᶜ)
+    σᶜᶠ = ifelse(hᶜᶠ == 0, one(grid), Hᶜᶠ / hᶜᶠ)
```

Comment:
why is this 1 for the case that hits the conditional?

Reply:
I was thinking that a zero `dz` might create some NaNs were we don't want them. This assumes that if the height is zero (which basically means that we do not have active cells in the column) we revert to a z-coordinate representation

Reply:
I think it's good to define that behavior. Maybe we will learn more about the implications when we have tests. Should we make a comment about it?

Reply:
Sure I am going to resee al the comments